File: backend/app/schemas/movie.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class SubtitleResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: int
    language: str
    format: str
    is_default: bool
    url: str


class MovieResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: int
    title: str
    year: int | None
    duration_seconds: int | None
    description: str | None
    genre: str | None
    poster_url: str
    backdrop_url: str
    stream_url: str
    file_extension: str
    file_size: int
    date_added: datetime
    video_width: int | None
    video_height: int | None
    quality: str | None
    video_codec: str | None
    audio_codec: str | None
    audio_channels: int | None
    bitrate: int | None
    frame_rate: str | None
    container_format: str | None
    subtitle_count: int
    subtitles: list[SubtitleResponse]


class ScanResponse(BaseModel):
    status: str
    scanned_files: int
    added: int
    updated: int
    removed: int
    ignored: int
    missing_directories: list[str]

File: android-tv/app/src/main/java/com/jasond/homeflix/data/model/Movie.kt
package com.jasond.homeflix.data.model

data class Subtitle(val id: Long, val language: String, val format: String, val isDefault: Boolean)

data class Movie(
    val id: Long, val title: String, val year: Int?, val durationSeconds: Long?,
    val description: String?, val genre: String?, val posterUrl: String?, val backdropUrl: String?,
    val fileExtension: String?, val fileSize: Long?, val dateAdded: String?,
    val videoWidth: Int?, val videoHeight: Int?, val quality: String?,
    val videoCodec: String?, val audioCodec: String?, val audioChannels: Int?,
    val subtitleCount: Int, val subtitles: List<Subtitle>,
)
